Update exporters documentation for JS

svrnm opened the pull request:
Major rework of the [Exporters](https://opentelemetry.io/docs/instrumentation/js/exporters/) page for JavaScript. 

The goal is once again to create a blueprint that then can be reused for other languages.

cartermp reviewed (changes requested):
Overall looks good, some questions though

svrnm commented on content/en/docs/instrumentation/js/exporters.md:
```diff
@@ -14,21 +14,137 @@ can learn how to setup exporters following the
+. Later you may want to [configure the collector](/docs/collector/configuration)
+to send your telemetry to your observability backend.
+
+#### Jaeger (Traces)
```
Similar to the getting started or manual instructions I want to rewrite that page in a way that it is self-enclosed. Here a part for that is having a "backend" (collector/jaeger/prom), ready to use and to try out what is described for the different protocols.
Because of that, my line of thinking (for both Jaeger and Prometheus below), was that as a first step people might like to setup a sample environment to send their traces/metrics to (collector is an option, but they like to see that maybe), also this is for OTLP and not for jaeger exporter (which is deprecated anyhow) and prometheus exporter.

Taking another look at that myself, I think there is an option to change this to something like

- Collector & Backend Setup
  - Collector: instructions for the logging collector, but with receivers for all protocols on this page (otlp, zipkin, prometheus)
  - Jaeger: instructions to setup jaeger to accept OTLP and Zipkin
  - Prometheus: instructions to setup prometheus to accept OTLP and "default" prometheus
- OTLP
- Console
- Jaeger
- Prometheus
- Zipkin
- Other available exporters
- Batching spans

In a second step I want to take the top part on setting up the collector/backend and extract it to a common section for all languages once again

wdyt?

svrnm replied:
[3ed156c](https://github.com/open-telemetry/opentelemetry.io/pull/3355/commits/3ed156cae4746f4c9e2f089f00f4c03708e6842e) changed that, please take another look

svrnm commented on content/en/docs/instrumentation/js/exporters.md:
````diff
@@ -14,21 +14,137 @@ can learn how to setup exporters following the
+  jaegertracing/all-in-one:latest
+```
+
+#### Prometheus (Metrics)
````
see https://github.com/open-telemetry/opentelemetry.io/pull/3355#discussion_r1356389812

svrnm commented:
I updated the content once again, I moved the setup of an example backend to the top of the page, such that it works for all potential protocols. I also applied some additional fixes. PTAL!

cartermp commented on content/en/docs/instrumentation/js/exporters.md:
```diff
@@ -14,21 +14,181 @@ can learn how to setup exporters following the
+[OpenTelemetry Collector](/docs/collector/), [Jaeger](https://jaegertracing.io)
+and [Prometheus](https://prometheus.io) to get you started quickly.
+
+### Collector (Traces, Metrics, Logs)
```
Hmmm. This section still doesn't sit right with me. It feels like the spirit of this section is to figure out how to export OTLP from the JS SDK, but here we're configuring a collector to scrape a prometheus endpoint and accept Zipkin data, when neither of those things are documented for the JS SDK. I think I'd prefer to have this be a minimal collector that just lets you verify your OTLP data is getting exported correctly.

svrnm replied:
> Hmmm. This section still doesn't sit right with me. It feels like the spirit of this section is to figure out how to export OTLP from the JS SDK, [...]

That's not the case anymore, I moved that section up in the hierarchy to be a "top level" section giving people the opportunity to setup a sample backend that works with all (most?) exporters coming later on this page:

<img width="271" alt="Screenshot 2023-10-20 at 12 43 43" src="https://github.com/open-telemetry/opentelemetry.io/assets/1519757/c91ec9eb-bed9-43e5-9b5c-3552c737fedd">

Ultimately I would move this out into a snippet again and make it available for all the languages.

My goal is making it easy for people to go through that documentation and if they don't have a backend yet or sending it to their production backend is complicated, they can quickly get started here

If this is too much for that page, the alternative is to move this out to some other page that can be referenced from here (not sure yet where to put it, but that's something we can figure out)

cartermp reviewed (approved):
I like the last changes a bunch. This looks great!

cartermp commented:
@open-telemetry/javascript-approvers can you take a look?

chalin commented on content/en/docs/instrumentation/js/exporters.md:
```diff
@@ -14,21 +14,100 @@ can learn how to setup exporters following the
+[configure the collector](/docs/collector/configuration) to send your telemetry
+to your observability backend.
+
+### Dependencies {#otlp-dependencies}
```
Good call on the anchor id.

svrnm replied:
thx, I hope I have them all right now :-)

chalin commented on content/en/docs/instrumentation/js/exporters.md:
```diff
@@ -14,21 +14,100 @@ can learn how to setup exporters following the
+Start by installing the respective exporter packages as a dependency for your
+project:
+
+{{< tabpane text=true langEqualsHeader=false >}} {{% tab "HTTP/Proto" %}}
```
```suggestion
{{< tabpane text=true >}} {{% tab "HTTP/Proto" %}}
```

chalin commented on content/en/docs/instrumentation/js/exporters.md:
````diff
@@ -14,21 +14,100 @@ can learn how to setup exporters following the
+  @opentelemetry/exporter-metrics-otlp-http
+```
+
+{{% /tab %}} {{% tab "gRPC" %}}
````
```suggestion
{{% /tab %}} {{% tab gRPC %}}
```

chalin reviewed (approved):
Only made a quick pass for potential copyedits. LGTM on that front.

cartermp commented:
@open-telemetry/javascript-approvers ping ping! I'd love to get this merged.

pichlermarc reviewed (approved):
This looks great. Thank you for working on this!
